### PR TITLE
Refactor controller.menu (Replace terminal-menu with inquirer.prompt). Closes gh-355

### PR DESCRIPTION
**Pull request by rwaldron**
See #355

Previously: 

After refactor:

### CODE REVIEW

File: lib/controller.js [hunk 1]
```diff
@@ -136,12 +138,43 @@ Tessel.get = function(opts) {
...
+                        map[display] = i;
+
+                        return display;
+                      })
+                    },
+                    translate: function(answer) {
+                      return tessels[map[answer.selected]];
+                    }
+                  }).then(function(tessel) {
+                    if (!tessel) {
```
Comment (johnnyman727) [inline lib/controller.js]: If a single Tessel was chosen, we should close all the non-selected Tessels.

**Comment by johnnyman727** [comment]
@rwaldron it looks good! I made a PR onto your PR to fix the tests and attend to my comment above about closing unused connections: https://github.com/rwaldron/t2-cli/pull/1

File: lib/controller.js (hunk 1 above)
Comment (johnnyman727) [inline lib/controller.js]: Nvm, that's handled in `logAndClose`.

**Comment by Student007** [comment]
@rwaldron @johnnyman727 :cry: my nice red menu :crying_cat_face: 

But if its better it is :+1: 

Once when I was thinking about a lot of different menu solutions I also took a look onto inquirer as a lot others like my favorite [Blessed](https://github.com/chjj/blessed#scrollablebox-from-box).
Also I thought about using nodes readline as you can see in this [example](https://nodejs.org/api/readline.html#readline_example_tiny_cli).

**The reason I decided to go with `terminal-menu` in version `2.x` had been to create a cooperate identity matching solution (what feels like to promise `t2-cli` to be a user friendly tool)**. Blessed would be to much but same as [terminal-kit](https://github.com/cronvel/terminal-kit). All other solutions I took a look onto either were beta or experimental.

There is also a fork of `terminal-menu` what provides selection and switches (see: [terminal-menu-program](https://github.com/Pomax/terminal-menu-program)). If the goal is to use selections this would be an option if the need of brand conform cooperate identity is needed.

About the problem of look and feel like @rwaldron shows up, it could be simple be solved by calling `controller.menu.clear().then(function(){// go forwards})`

But as I said: If majority is for replacing terminal-menu I didn't have a problem with this - the best solution should be in - not the hardest work :smile: 

Maybe @Frijol also would put her two cents in (as a marketing professional).

**Comment by johnnyman727** [comment]
@Student007 don't be :crying_cat_face:! Be :joy_cat:! Every contribution builds on the work of the previous. Without your help, it would have been much more work to improve on the menu experience. It's important to get used to your code being overwritten because, in a successful project, that will happen all the time.

**Pull request merged**